inference-engine/src/transformations/src/transformations/smart_reshape/proposal_scales_stridedslice.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <transformations/smart_reshape/proposal_scales_stridedslice.hpp>
+
+#include <ngraph/ngraph.hpp>
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/pattern/matcher.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ProposalScales, "ProposalScales", 0);
+
+ngraph::pass::ProposalScales::ProposalScales() {
+    auto parameter_label = ngraph::pattern::wrap_type<opset5::Parameter>({}, [](const Output<Node> &output) {
+        const auto & shape = output.get_partial_shape();
+        return shape.rank().is_static() && shape.rank().get_length() == 2 && shape[1].is_static() && (shape[1].get_length() == 3 || shape[1].get_length() == 4);
```

Comment:
That's an amazing case!
I would see our future like this:
```
pattern::all_of(
        pattern::rank_is_equal_to(2),
        pattern::any_of(pattern::dim_is_equal_to(1, 3),
                        pattern::dim_is_equal_to(1, 4))
);
```

What do you think?

Reply:
I would love to allow batch to be dynamic or static but any number, not only 1, like this 
`pattern::dim_is_equal_to(pattern::dim::any(), 3)`